src/Serilog/Configuration/LoggerSinkConfiguration.cs

```diff
@@ -205,27 +207,37 @@ internal LoggerSinkConfiguration(LoggerConfiguration loggerConfiguration, Action
             if (wrapSink == null) throw new ArgumentNullException(nameof(wrapSink));
             if (configureWrappedSink == null) throw new ArgumentNullException(nameof(configureWrappedSink));
 
-            void WrapAndAddSink(ILogEventSink sink)
-            {
-                bool sinkIsDisposable = sink is IDisposable;
-
-                ILogEventSink wrappedSink = wrapSink(sink);
-
-                if (sinkIsDisposable && !(wrappedSink is IDisposable))
-                {
-                    SelfLog.WriteLine("Wrapping sink {0} does not implement IDisposable, but wrapped sink {1} does.", wrappedSink, sink);
-                }
-                loggerSinkConfiguration.Sink(wrappedSink, restrictedToMinimumLevel, levelSwitch);
-            }
+            var sinksToWrap = new List<ILogEventSink>();
 
+            var capturingConfiguration = new LoggerConfiguration();
             var capturingLoggerSinkConfiguration = new LoggerSinkConfiguration(
-                loggerSinkConfiguration._loggerConfiguration,
-                WrapAndAddSink,
+                capturingConfiguration,
+                sinksToWrap.Add,
                 loggerSinkConfiguration._applyInheritedConfiguration);
 
+            // `WriteTo.Sink()` will return the capturing configuration; this ensures chained `WriteTo` gets back
+            // to the capturing sink configuration, enabling `WriteTo.X().WriteTo.Y()`.
+            capturingConfiguration.WriteTo = capturingLoggerSinkConfiguration;
+
             configureWrappedSink(capturingLoggerSinkConfiguration);
 
-            return loggerSinkConfiguration._loggerConfiguration;
+            if (!sinksToWrap.Any())
```

Comment:
Might as well grab `Count` as it's used 3 lines down

Reply:
Done :-)